src/main/java/org/opensearch/security/resources/ResourceSharingIndexHandler.java

```diff
@@ -493,64 +423,50 @@ public void onFailure(Exception e) {
      *                        "backend_roles": ["backend_role1"]
      *                        }
      *                        }
-     * @param isAdmin         Boolean indicating whether the user requesting to revoke is an admin or not
      * @param listener        Listener to be notified when the operation completes
      * @throws RuntimeException if there's an error during the update operation
      */
-    @SuppressWarnings("unchecked")
-    public void updateSharingInfo(
-        String resourceId,
-        String resourceIndex,
-        String requestUserName,
-        ShareWith shareWith,
-        boolean isAdmin,
-        ActionListener<ResourceSharing> listener
-    ) {
+    public void share(String resourceId, String resourceIndex, ShareWith shareWith, ActionListener<ResourceSharing> listener) {
         StepListener<ResourceSharing> sharingInfoListener = new StepListener<>();
 
         // Fetch resource sharing doc
         fetchSharingInfo(resourceIndex, resourceId, sharingInfoListener);
 
         // build update script
         sharingInfoListener.whenComplete(sharingInfo -> {
-            // Check if user can share. At present only the resource creator and admin is allowed to share the resource
-            if (!isAdmin && sharingInfo != null && !sharingInfo.getCreatedBy().getUsername().equals(requestUserName)) {
-
-                LOGGER.error("User {} is not authorized to share resource {}", requestUserName, resourceId);
-                listener.onFailure(
-                    new OpenSearchStatusException(
-                        "User " + requestUserName + " is not authorized to share resource " + resourceId,
-                        RestStatus.FORBIDDEN
-                    )
-                );
+            if (sharingInfo == null) {
+                LOGGER.debug("No sharing record found for resource {}", resourceId);
+                listener.onResponse(null);
                 return;
             }
-
             for (String accessLevel : shareWith.accessLevels()) {
                 Recipients target = shareWith.atAccessLevel(accessLevel);
-                assert sharingInfo != null;
+
                 sharingInfo.share(accessLevel, target);
             }
 
             String resourceSharingIndex = getSharingIndex(resourceIndex);
             try (ThreadContext.StoredContext ctx = threadPool.getThreadContext().stashContext()) {
-                assert sharingInfo != null;
                 IndexRequest ir = client.prepareIndex(resourceSharingIndex)
                     .setId(sharingInfo.getResourceId())
-                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE) // TODO check this policy
```

Comment:
nit: remove this TODO, any index into a system index should have refresh immediate

Reply:
I agree. I think i added this todo while I was debugging Windows failures where I tried WAIT_UNTIL. But eventually those failures turned out to be sync vs async nature issue of ResourceAccessEvaluator.